### DISCUSSION TITLE
Can I use snippet just to trigger vim command ?

### DISCUSSION THREAD

**Discussion by aca** [L3MON4D3/LuaSnip#917] [Q&A] [answered]
Hi,

There's command something like this `:lua vim.cmd.GoIfErr()`.

Which insert snippet 
```
if err != nil {
    return err
}
```

It's provided by another plugin (which executes shell command). 

Instead of executing command in vim or `inoremap`, I want to use luasnip to trigger the command. 

I tried this but doesn't work. 
```
ls.add_snippets("go", {
    s({ trig = "rerr", dscr = "iferr" }, {
        f(function()
            vim.cmd.GoIfErr()
        end, {}),
    }),
})
```

Wish I could get some help. Thanks.

**Comment by L3MON4D3** [comment, marked as answer]
TBH, I don't think this is a good idea, we assume that the buffer is not modified while inserting a snippet.
It might work if the functionNode returns a `""` (ie the snippet does not insert any text), but please don't count on it.

**Comment by leiserfg** [comment]
I think it's better to use an `iabbrev` for this use case.

**Comment by aca** [comment]
Ended up creating plugin with nvim-cmp, thanks for the reply.
If anyone who wants to trigger function, you might try https://github.com/aca/cmp-function
